EventFiltering/PWGHF/HFFilterHelpers.h

```diff
@@ -1693,14 +1693,26 @@ inline int16_t HfFilterHelper::isSelectedBachelorForCharmBaryon(const T& track,
     nSigmaPrTpc = getTPCSplineCalib(track, (track.sign() > 0) ? kPr : kAntiPr);
   }
 
-  if ((track.hasTPC() && std::fabs(nSigmaPiTpc) > mNSigmaTpcPiCharmBaryonBachelor) && (track.hasTOF() && std::fabs(nSigmaPiTof) > mNSigmaTofPiCharmBaryonBachelor)) {
-    CLRBIT(retValue, kPionForCharmBaryon);
-  }
-  if ((track.hasTPC() && std::fabs(nSigmaKaTpc) > mNSigmaTpcPiCharmBaryonBachelor) && (track.hasTOF() && std::fabs(nSigmaKaTof) > mNSigmaTofPiCharmBaryonBachelor)) {
-    CLRBIT(retValue, kKaonForCharmBaryon);
-  }
-  if ((track.hasTPC() && std::fabs(nSigmaPrTpc) > mNSigmaTpcPiCharmBaryonBachelor) && (track.hasTOF() && std::fabs(nSigmaPrTof) > mNSigmaTpcPiCharmBaryonBachelor)) {
-    CLRBIT(retValue, kProtonForCharmBaryon);
+  if (track.hasTOF()) {
+    if ((track.hasTPC() && std::fabs(nSigmaPiTpc) > mNSigmaTpcPiCharmBaryonBachelor) && (track.hasTOF() && std::fabs(nSigmaPiTof) > mNSigmaTofPiCharmBaryonBachelor)) {
+      CLRBIT(retValue, kPionForCharmBaryon);
+    }
+    if ((track.hasTPC() && std::fabs(nSigmaKaTpc) > mNSigmaTpcPiCharmBaryonBachelor) && (track.hasTOF() && std::fabs(nSigmaKaTof) > mNSigmaTofPiCharmBaryonBachelor)) {
+      CLRBIT(retValue, kKaonForCharmBaryon);
+    }
+    if ((track.hasTPC() && std::fabs(nSigmaPrTpc) > mNSigmaTpcPiCharmBaryonBachelor) && (track.hasTOF() && std::fabs(nSigmaPrTof) > mNSigmaTpcPiCharmBaryonBachelor)) {
+      CLRBIT(retValue, kProtonForCharmBaryon);
+    }
```

Comment:
thanks @apalasciano for implementing this improvement that we discussed. A super minor point: you can remove `track.hasTOF() &&` inside the `if` statements here, since now you check them only if the track matches TOF